plugins/reports/counter/classes/releases/CounterRelease4_1.inc.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+/**
+ * @file plugins/generic/sushiLite/classes/releases/CounterRelease4_1.inc.php
+ *
+ * Copyright (c) 2014 University of Pittsburgh
+ * Distributed under the GNU GPL v2 or later. For full terms see the file docs/COPYING.
+ *
+ * @class CounterRelease4_1
+ * @ingroup plugins_reports_counter
+ *
+ * @brief A COUNTER report, release 4.1
+ */
+import('plugins.reports.counter.classes.releases.CounterRelease');
+
+class CounterRelease4_1 extends CounterRelease {
+
+	/**
+	 * Constructor
+	 * @param string $parentPluginCategory
+	 * @param string $parentPluginName
+	 */
+	function CounterRelease4_1($parentPluginCategory, $parentPluginName) {
+		parent::CounterRelease($parentPluginCategory, $parentPluginName);
+	}
+	
+	/**
+	 * Construct a Reports result containing the provided performance metrics
+	 * @param $columns string|array column (aggregation level) selection
+	 * @param $filters array report-level filter selection
+	 * @param $orderBy array order criteria
+	 * @param $range null|DBResultRange paging specification
+	 * @see ReportPlugin::getMetrics for more details
+	 * @return string xml
+	 */
+	function getMetricsXML($columns = array(), $filters = array(), $orderBy = array(), $range = null) {
+		$reportItems = $this->getMetrics($columns, $filters, $orderBy, $range);
```

Comment:
This class is using a method (getMetrics) that's implemented only by the class that extends it. This is not a good design approach, the parent class should not know the details of the class that implement it.

One easy solution is to add the getMetrics() method as a method into the general report class, so it's clear that child classes should implement it. It could be abstract (which requires an implementation) or we can implement something by default and leave the other details to the child classes.